src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests_LifetimeAnnotation.cs

```diff
@@ -354,14 +355,52 @@ static void M()
 }";
             var comp = CreateCompilation(source);
             var expected =
-@"";
+@"void Program.<M>g__L|0_0(in System.Int32 i)
+    [LifetimeAnnotation(True, False)] in System.Int32 i
+";
+            CompileAndVerify(
+                source,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All),
+                symbolValidator: module =>
+                {
+                    Assert.Equal("System.Runtime.CompilerServices.LifetimeAnnotationAttribute", GetLifetimeAnnotationType(module).ToTestDisplayString());
+                    AssertLifetimeAnnotationAttributes(module, expected);
+                });
+        }
+
+        [Fact]
+        public void EmitAttribute_InferredDelegateParameters()
+        {
+            var source =
+@"ref struct R { }
+class Program
+{
+    static void Main()
+    {
+        var d1 = (scoped in int i) => { };
+        d1(0);
+        var d2 = (scoped R r) => new R();
+        d2(new R());
+    }
+}";
+            var comp = CreateCompilation(source);
+            var expected =
+@"void <>f__AnonymousDelegate0.Invoke(in System.Int32 value)
+    [LifetimeAnnotation(True, False)] in System.Int32 value
```

Comment:
I was slightly surprised we decided to implement these attributes on synthesized unspeakable methods. I thought since the runtime doesn't check these that it doesn't really make a difference.

Reply:
It's a good point that the runtime doesn't check these attributes, so we could probably drop the attributes on the synthesized methods. I think I'd prefer to keep the attributes though, unless we're comfortable ruling out future scenarios that might use this info.